src/coverage_reporter/parsers/clover_parser.cr

```diff
@@ -57,8 +57,12 @@ module CoverageReporter
             branches[line_number] << branch_hits
           end
 
-          hits = line_node.attributes["count"].content.to_u64
-          coverage[line_number] = hits
+          if line_type == "method"
+            coverage[line_number] = 1
+          else
+            hits = line_node.attributes["count"].content.to_u64
+            coverage[line_number] = hits
+          end
         end
```

Comment:
@mike-burns question:
Will this solution place the correct number of hits for the method on the line declaring the method (in the Coveralls source file report)? See the example I refer to below, pls.

I see that this code will solve the problem of lines declaring functions displaying as (relevant but) uncovered. It appears this will always at least make sure they are shown as covered (green) in Coveralls reports.

However, given your discovery that the number of hits for a **method** is taken to be equivalent to the highest number of hits for any of its (**statement**) lines, won't this miss the opportunity to show the correct number of hits for the method, and always just show (`1`)?

I can understand using this code as a failsafe to make sure lines declaring a method appear as covered, but it seems to me that the correct number of hits should be what we get from a correct XML report, which appears to be that highest number for the method's lines. 

Check out the example pertaining to the `AbstractArrayDeclarationSniff.php` file in [my comment](https://github.com/lemurheavy/coveralls-public/issues/1721#issuecomment-1851023472) on the [OP's public issue](https://github.com/lemurheavy/coveralls-public/issues/1721):

<img width="827" alt="Screenshot 2024-04-02 at 4 27 34 PM" src="https://github.com/coverallsapp/coverage-reporter/assets/918667/4ff99df5-1a56-46d0-9bc1-7e5d976c4318">

To be perfectly honest, I kind of forget the point of my comment there, but now I'm stuck on the fact that recent source file reports (like [this one](https://coveralls.io/builds/64474766/source?filename=PHPCSUtils%2FAbstractSniffs%2FAbstractArrayDeclarationSniff.php#L178), as opposed to [this older version from Aug 2023](https://coveralls.io/builds/62230525/source?filename=PHPCSUtils%2FAbstractSniffs%2FAbstractArrayDeclarationSniff.php#L178)) seem to show the correct hit count for the method. in the case of Line 178, `52`.

Isn't that what a correct clover.xml report should show as the hit count for the method? And isn't that what we want to show in our reports?

Forgive me for being rusty on this one, I may have missed something.

Reply:
> won't this miss the opportunity to show the correct number of hits for the method, and always just show (`1`)?

This really comes down to: what number do we want to show for line 178 in that screenshot? What is "correct", in this case?

- `52`, the highest number of any line within the method (so `0` for an untested method).
- `max(52, 1)`, the highest number of any line within the method (or `1` for an untested method, meaning that line 178 is OK).
- `1`, simply indicating that the method was defined.

As a base case, what numbers do we want for this method definition:

```php
public function someNoOpMethod()
{
}
```

Do we want to say that the `public function someNoOpMethod()` line was never run? No amount of test coverage can change that -- calling the method executes zero statements, so the method definition line is still considered `0`.

I'm happy to go with your decision -- you might have more context than I do around what these numbers are used for.